src/foam/core/CompoundContextAgent.java

```diff
@@ -0,0 +1,55 @@
+/**
+ * @license
+ * Copyright 2019 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.core;
+
+import java.util.ArrayList;
+
+/**
+ * A ContextAgent which is itself an Agency.
+ * When ContextAgents are submit()'ed to the service it queues them
+ * but doesn't execute them until it is itself executed.
+ **/
+public class CompoundContextAgent
+  implements ContextAgent
+{
+  public ArrayList<Runnable> agents_ = new ArrayList();
+
+  public void execute(X x) {
+    for ( Runnable agent : agents_ ) {
+      try {
+        agent.run();
+      } catch (Throwable t) {
+        throw t;
```

Comment:
Catching and rethrowing is the same as doing nothing. Should add to a CompoundException.